vsphere_plugin_common/__init__.py

```diff
@@ -2192,7 +2193,8 @@ def host_cpu_thread_usage_ratio(self, host, vm_cpus):
 
         total_assigned = vm_cpus
         for vm in host.vm:
-            total_assigned += vm.summary.config.numCpu
+            if isinstance(vm.summary.config.numCpu, (int, long, float, complex)):
+                total_assigned += vm.summary.config.numCpu
```

Comment:
If it's not the right type shouldn't we at least log a warning, if not raising an error?

Also, I don't think we need to allow noninteger CPU counts, and certainly not complex numbers.

The same likely applies to the memory.

Reply:
probably `try: float(); except ValueError: ...` is the nicer thing to do instead of the isinstance check

Reply:
Also (from speaking to @tehasdf), it would probably be better to attempt to cast to float (or more likely int- I'm not sure what the API will accept) wrapped in a try/except rather than checking isinstance.

Reply:
Sorry for the delay. I've made changes according to Your suggestion. @geokala Please check new commit, Now I'm casting it to the integer as proposed. According to this api reference (https://www.vmware.com/support/developer/converter-sdk/conv50_apireference/vim.vm.Summary.ConfigSummary.html) those values are integers. In this case I think warning is enough.

Reply:
In this case, a warning means that if a user typos the value, a warning will be logged but the VM will be deployed.
Is this desirable behaviour?

I'm just thinking in the case that this results in something being deployed and the warning being lost in all the other messages that get emitted during an install operation, it might result in someone thinking their application server was deployed successfully and then finding out that it had half the RAM/CPUs they thought it did later.

Given that failure mode, should this be a warning or a NonRecoverableError?

Reply:
What about replace:
```python
int(vm.summary.config.numCp)
total_assigned += vm.summary.config.numCpu
```
to
```python
total_assigned += int(vm.summary.config.numCpu)
```